### ISSUE TITLE
ability to modify the color

### ISSUE THREAD

**Issue by syrakozz** [ask-fm/AFMInfoBanner#3]
can you plz add the abiliy to modfiy the background color

**Comment by artworkad** [comment]
:+1:

**Comment by karpelcevs** [comment]
Hi!

There are `errorBackgroundColor` and `infoBackgroundColor` properties that support `UIAppearance`. Could you specify what do you mean by modifying background color?

**Comment by ndmeiri** [comment]
@coverback Would you consider reopening this? I would like the ability to show banners with more than two possible background colors.

**Comment by crazyjooe** [comment]
@ndmeiri there is option to modify background colors via previously described properties. Did you try to use them? 
The easiest way to set them for the whole project is via UIAppearance

**Comment by ndmeiri** [comment]
@crazyjooe Yes, I understand that I can change the colors for the banner styles. But since there are only 2 banner styles (`.Info` and `.Error`), I can only ever show two different colors of banners. Of course, I could use `UIAppearance` to change the colors between banner presentations, but this is unnecessarily complicated and not the intended use case for `UIAppearance`.

**Comment by karpelcevs** [comment]
Hi @ndmeiri! This makes sense. Let us evaluate the change and we'll get back to you.
Should be pretty easy.
How would you like to see the API for that? Like additional sets of predefined styles or same style, but passing background and text colours per each instance?

**Comment by karpelcevs** [comment]
And just in case, right now you can create an instance of the banner with `init` or `show` methods and then adjust the colour of the created banner. To reopen and work on this it is important for us to understand, why create+adjust colour isn't working for you.

**Comment by ndmeiri** [comment]
Hi @coverback, thanks for the info. I'm trying to create an instance with `init`, but I can't change the background color or text color because the relevant properties are private. Could you please show me an example of creating an instance with `init` with custom background and text colors? Thanks!

**Comment by ndmeiri** [comment]
I was thinking we could have a protocol `AFMBannerStyler` as follows (in Swift):

``` Swift
@objc protocol AFMBannerStyler {
    optional var bannerBackgroundColor: UIColor { get }
    optional var bannerTextColor: UIColor { get }
    optional var bannerFont: UIFont { get }
}
```

And then the `show` family of class methods would use this protocol instead (again, in Swift)

``` Swift
public class func showWithText(text: String!, style: AFMBannerStyler, andHideAfter timeout: NSTimeInterval) -> Self!
public class func showWithText(text: String!, style: AFMBannerStyler, animated: Bool) -> Self!
public class func showAndHideWithText(text: String!, style: AFMBannerStyler) -> Self!
```

`AFMInfoBanner` would then use the `AFMBannerStyler` to apply the banner style. If the object that conforms to `AFMBannerStyler` does not implement any of the optional computed properties, `AFMInfoBanner` uses a default value for that property.

**Comment by ndmeiri** [comment]
But the `init` pattern is probably more simple

**Comment by crazyjooe** [comment]
@ndmeiri we just released 1.2 version which allows to customise banner after init.

``` objective-c
AFMInfoBanner *banner = [[AFMInfoBanner alloc] init];
banner.customBackgroundColor = [UIColor yellowColor];
banner.customTextColor = [UIColor redColor];
[banner show:YES];
```

Custom background and text colors have priority over style colors.

**Comment by ndmeiri** [comment]
@crazyjooe This is great. I'll integrate it in my project as soon as I have a chance. Thank you for the update.

**Comment by ndmeiri** [comment]
@crazyjooe So when I show the banner, does it matter which `AFMInfoBannerStyle` I pass? From what you said above, I believe it shouldn't matter. But then the perceived behavior of the line of code that shows a banner is different from the actual behavior of that line of code. Do you know what I mean? i.e. Why should I have to pass in a style if it's going to get overridden?

**Comment by ndmeiri** [comment]
@crazyjooe So when I show the banner, does it matter which `AFMInfoBannerStyle` I pass? From what you said above, I believe it shouldn't matter. But then the perceived behavior of the line of code that shows a banner is different from the actual behavior of that line of code. Do you know what I mean? i.e. Why should I have to pass in a style if it's going to get overridden?

**Comment by ndmeiri** [comment]
@crazyjooe What is the best way to automatically hide the banner after showing it with `show()` (in Swift)? Since I can no longer use the class methods for showing and hiding automatically, am I forced to use a `showCompletionBlock`?

**Comment by crazyjooe** [comment]
@ndmeiri we released 1.2.1 version which has instance method `- (void)showAndHideAfter:(NSTimeInterval)timeout animated:(BOOL)animated;`.

**Comment by ndmeiri** [comment]
@crazyjooe Excellent. Thank you for the update.
